vscode/microsoft-kiota/package.json

```diff
@@ -461,24 +461,28 @@
     "package": "webpack --mode production --devtool hidden-source-map",
     "compile-tests": "tsc -p . --outDir out",
     "watch-tests": "tsc -p . -w --outDir out",
-    "pretest": "npm run compile-tests && npm run compile && npm run lint",
+    "pretest": "npm run compile && npm run compile-tests && npm run lint",
     "lint": "eslint",
-    "test": "node ./out/test/runTest.js"
+    "test": "vscode-test",
+    "test-with-coverage": "npm run pretest && vscode-test --coverage"
   },
   "devDependencies": {
     "@stylistic/eslint-plugin-ts": "^2.9.0",
     "@types/adm-zip": "^0.5.5",
     "@types/mocha": "^10.0.9",
     "@types/node": "22.x",
+    "@types/sinon": "^17.0.3",
     "@types/vscode": "^1.94.0",
     "@typescript-eslint/eslint-plugin": "^8.11.0",
     "@typescript-eslint/parser": "^8.11.0",
+    "@vscode/test-cli": "^0.0.10",
     "@vscode/test-electron": "^2.4.1",
     "eslint": "^9.13.0",
     "glob": "^11.0.0",
     "mocha": "^10.7.3",
+    "sinon": "^19.0.2",
     "ts-loader": "^9.5.1",
-    "typescript": "^5.6.3",
+    "typescript": "^5.5.4",
```

Comment:
not sure why we're downgrading TS here? looks like a merge issue

Reply:
We initially had an issue with typescript 5.6.2 as reported [here](https://github.com/isaacs/node-lru-cache/issues/352) but I will upgrade to 5.6.3 and see if the issue is now fixed given the issue is closed or a better alternative available than to edit the lru_cache file (which is a dirty local option in my opinion)